src/serialization/pair.h

```diff
@@ -42,12 +44,31 @@ namespace serialization
       return ::do_serialize(ar, e);
     }
 
-    template <typename Archive>
-    bool serialize_pair_element(Archive& ar, uint64_t& e)
-    {
-      ar.serialize_varint(e);
-      return true;
+#define PAIR_VARINT_SERIALIZER(type) \
+    template <typename Archive> \
+    bool serialize_pair_element(Archive& ar, type& e) \
+    { \
+      if (serialization_size_t_backward_compat && !typename Archive::is_saving() && !std::is_same<uint64_t, type>()) \
```

Comment:
Yes, old mac (and possibly any other platform where size_t is distinct from uint64_t). I'm unsure about the switch from size_t to uint64_t. I guess it's a bit more compatible between 32 and 64 bit archs, but otherwise would not matter. Old hot/cold data from mac can be deleted and recreated.

Reply:
I was commenting on Macs being unable to read older files they produced after the `size_t` -> `uint64_t` change. This seems to be limited to data exported for offline usage, but wanted to make sure this was limited to cases that were unlikely to cause many problems for users (like the wallet file itself).